dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/plugins/delegates/MediaTypeHeaderDelegate.java

```diff
@@ -0,0 +1,190 @@
+package org.jboss.resteasy.plugins.delegates;
+
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+import org.jboss.resteasy.util.HeaderParameterParser;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.RuntimeDelegate;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+public class MediaTypeHeaderDelegate implements RuntimeDelegate.HeaderDelegate
+{
+   public static final MediaTypeHeaderDelegate INSTANCE = new MediaTypeHeaderDelegate();
+
+   private static Map<String, MediaType> map = new ConcurrentHashMap<String, MediaType>();
+   private static Map<MediaType, String> reverseMap = new ConcurrentHashMap<MediaType, String>();
+   private static final int MAX_MT_CACHE_SIZE =
+                   // Liberty Change
```

Comment:
this has been fixed in RESTEasy, but not yet GA - should be in 4.5.7 when it comes out. maybe mention that in the comment so we know that we can delete this overlay once we pull in 4.5.7 or newer?

Reply:
If this has been fixed in the resteasy master branch, would it be better to build a snapshot version from there and put it in DHE ... rather than creating a new overlay!?

Reply:
Updated the comment. @tevans78 We plan to upgrade to the newer version of Resteasy before release, these overlay changes are are to enable us to do rapid development. This is currently how we override our CXF implementation for ```jaxrs-2.0``` and ```jaxrs-2.1```

We discussed having our own CXF fork and pulling that into OpenLiberty on today's design issues call. I assume we'll have a similar discussion about Resteasy in the future.

Reply:
Rapid development now but at what service cost if the overlays are never removed ... which is exactly what has happened with CXF. Your aim should be to avoid a fork or overlay unless there is no other choice. In this case there is a choice, the change is already in RESTEasy.

Reply:
I don't disagree, but I think this conversation is beyond the scope of this PR.